rfc/text/advanced/ap_10_rpc_progressive_calls.md

```diff
@@ -1,5 +1,250 @@
+
+{align="left"}
+     ,------.                  ,------.                 ,------.
+     |Caller|                  |Dealer|                 |Callee|
+     `--+---'                  `--+---'                 `--+---'
+        |     CALL (progress)     |                        |
+        | ----------------------> |                        |
+        |                         | INVOCATION (progress)  |
+        |                         | ---------------------->|
+        |     CALL (progress)     |                        |
+        | ----------------------->|                        |
+        |                         | INVOCATION (progress)  |
+        |                         | ---------------------->|
+        |                         |                        |
+        |           ...           |          ...           |
+        |                         |                        |
+        |     CALL (final)        |                        |
+        | ----------------------->|                        |
+        |                         |   INVOCATION (final)   |
+        |                         | ---------------------->|
+        |                         |                        |
+        |                         | YIELD (final) or ERROR |
+        |                         | <----------------------|
+        |                         |                        |
+        | RESULT (final) or ERROR |                        |
+        | <-----------------------|                        |
+     ,--+---.                  ,--+---.                 ,--+---.
+     |Caller|                  |Dealer|                 |Callee|
+     `------'                  `------'                 `------'
+
+As `progressive` call is still the same call *Callee* must send the same `Request|id` with every `CALL` and *Dealer*
+must also send the same `Request|id` with every `INVOCATION` to the *Callee*.
+
+
+The following message flow illustrates a progressive call when *Callee* starts sending progressive call 
+results immediately. Please note that `YIELD` messages don't need to be matched with `CALL`/`INVOCATION` messages. 
+E.g. *Caller* can send a few `CALL` messages before starting to receive `RESULTS`, they do not need to be correlated. 
+
+{align="left"}
+     ,------.                  ,------.                 ,------.
+     |Caller|                  |Dealer|                 |Callee|
+     `--+---'                  `--+---'                 `--+---'
+        |     CALL (progress)     |                        |
+        |------------------------>|                        |
+        |                         | INVOCATION (progress)  |
+        |                         |----------------------->|
+        |                         |                        |
+        |                         |   YIELD (progress)     |
+        |                         |<-----------------------|
```

Comment:
Is it possible to return ERROR early if the caller knows that the request cannot be processed?

Reply:
Yes, of course! If for some reason `Callee` understand that it can not process incoming `INVOCATION` — it can immediately send `ERROR` message. Thats ok.

Reply:
Yep, and should `Callee` decide to answer with `ERROR`, that concludes that invocation (no more messages from the `Callee`).

Btw: did we discuss combination of progressive calls & shared registrations for such a case already? What I mean is: assume there are 10 Callees with a shared registration of procedure P, and one of the Callees returns an ERROR while the progressive call is still ongoing, the *Router* MUST interrupt all other 9 Callees (by sending them an `INTERRUPT`), _and_ immediately send back `ERROR` to the original Caller

Reply:
I think we should clarify this in spec... So i've created #418